controller/src/controller/main_systems/system_monitor.py

```diff
@@ -143,26 +144,51 @@ async def _update_system_status_special_cases(self) -> None:
             await self._system_state_manager.update(system_status_dict)
 
     async def _push_system_status_update(self, update_details: ReadOnlyDict) -> None:
-        if status_update_details := {
+        status_update_details = {
             status_name: new_system_status
-            for status_name in ("system_status", "stimulation_protocols_running", "in_simulation_mode")
+            for status_name in ("system_status", "stimulation_protocol_statuses", "in_simulation_mode")
             if (new_system_status := update_details.get(status_name))
-        }:
-            logger.info(f"System status update: {status_update_details}")
-            # want to have system_status logged as an enum, so afterwards need to convert it to a string so it can be json serialized later
-            if system_status := status_update_details.get("system_status"):
-                status_update_details["system_status"] = str(system_status.value)
+        }
+        if not status_update_details:
+            return
+
+        logger.info(f"System status update: {status_update_details}")
+
+        # starting/stopping states should be logged, but don't need to be sent to the UI, so remove them
+        try:
+            new_stim_statuses = status_update_details.pop("stimulation_protocol_statuses")
+        except KeyError:
+            pass
+        else:
+            if not (
+                # if either of these are present then assume stim will either start or stop on every well soon so wait for that to happen before sending an update
+                StimulationStates.STARTING in new_stim_statuses
+                or StimulationStates.STOPPING in new_stim_statuses
+            ):
+                status_update_details["stimulation_protocols_running"] = [
+                    stim_status == StimulationStates.RUNNING for stim_status in new_stim_statuses
+                ]
+
+        # this will happen if the only update was starting/stopping stim, in which case nothing needs to be sent to the UI
+        if not status_update_details:
```

Comment:
is this also supposed to be on line 152?

Reply:
Yeah, could change it later but right now it's so that it logs all stim status changes including the transitions but doesn't send the transitions to the UI